### ISSUE TITLE
Debug mode information/legend

### ISSUE THREAD

**Issue by slosumo** [KilledByAPixel/LittleJS#19]
Would appreciate an additional legend to be added to debug overlay explaining the various colors/squares:
![image](https://user-images.githubusercontent.com/10269798/144053308-3db1892f-8436-47c6-98ef-c3dd0295e95e.png)

I assume one of the blues is showing world pos, 1 is showing render pos of the tile or something like that? I could decipher it but a quick glimpse at what these things mean would be great to haves!

**Comment by KilledByAPixel** [comment]
I will look into it, the debug mode is still kind of in flux, I have some better ideas on how to do it now.

**Comment by KilledByAPixel** [comment]
I improved the debug system!

I added toString to the EngineObject class which is used in the debug display. By default this looks similar to the screenshot above but can be easily extended to better suite the needs of any game.

The coloring is still the same for now. It's a basically a physics test from the camera point to the nearest object. If there is a collision the line is red and it shows the collision point. It also shows information about the mouse point and the nearest object.
